ur_controllers/src/gpio_controller.cpp

```diff
@@ -263,22 +263,23 @@ bool GPIOController::setIO(ur_msgs::srv::SetIO::Request::SharedPtr req, ur_msgs:
 bool GPIOController::setSpeedSlider(ur_msgs::srv::SetSpeedSliderFraction::Request::SharedPtr req,
                                     ur_msgs::srv::SetSpeedSliderFraction::Response::SharedPtr resp)
 {
-  if (req->speed_slider_fraction >= 0 && req->speed_slider_fraction <= 1.0) {
+  if (req->speed_slider_fraction >= 0.01 && req->speed_slider_fraction <= 1.0) {
     RCLCPP_INFO(node_->get_logger(), "Setting speed slider to %.2f%%.", req->speed_slider_fraction * 100.0);
     // reset success flag
-    command_interfaces_[CommandInterfaces::SCALING_ASYNC_SUCCESS].set_value(2.0);
+    command_interfaces_[CommandInterfaces::TARGET_SPEED_FRACTION_ASYNC_SUCCESS].set_value(2.0);
     // set commanding value for speed slider
-    command_interfaces_[CommandInterfaces::SPEED_SCALING_CMD].set_value(
+    command_interfaces_[CommandInterfaces::TARGET_SPEED_FRACTION_CMD].set_value(
         static_cast<double>(req->speed_slider_fraction));
 
-    while (command_interfaces_[CommandInterfaces::SCALING_ASYNC_SUCCESS].get_value() != 2.0) {
+    while (command_interfaces_[CommandInterfaces::TARGET_SPEED_FRACTION_ASYNC_SUCCESS].get_value() != 2.0) {
```

Comment:
Please define use constants for numbers that have a specific meaning.

Reply:
Changed it so that the shared cmd variable contains a NaN if no new data is available and capsuled it into a static constant

Reply:
Ok still have to have a look at what @livanov93 intended with the async communication at this point.

Reply:
Exchanged 2.0 value as well and removed a race condition between controller and hardware interface in the async check